libsolidity/parsing/Parser.cpp

```diff
@@ -2076,16 +2078,39 @@ optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> cons
 	for (auto const& [start, end]: sequencesToSearch)
 	{
 		smatch match;
-		if (regex_search(start, end, match, licenseRegex))
+		auto words_begin =
+			std::sregex_iterator(start, end, licenseRegex);
+		auto words_end = std::sregex_iterator();
+
+		for (std::sregex_iterator i = words_begin; i != words_end; ++i)
 		{
-			string license{boost::trim_copy(string(match[1]))};
-			if (!license.empty())
+			std::smatch match = *i;
+			if (!match.empty())
+			{
+				string license{string(match[1])};
```

Comment:
Does it actually need a this extra string conversion?
```suggestion
				string license = match[1];
```